sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoaderTemp.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import com.nvidia.spark.GpuCachedBatchSerializer
+import com.nvidia.spark.rapids.delta.DeltaProbe
+import com.nvidia.spark.rapids.iceberg.IcebergProvider
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.rapids.{AdaptiveSparkPlanHelperShim, ExecutionPlanCaptureCallbackBase}
+
+// TODO contains signatures that cannot yet be moved to sql-plugin-api
+object ShimLoaderTemp extends Logging {
```

Comment:
> // TODO contains signatures that cannot yet be moved to sql-plugin-api
is there a plan to move them, and then remove the `Temp` side of it? I guess I am unclear on how apis are supposed to move to sql-plugin-api

Reply:
Yes, everything here accessed via the code before the plugin is initialized, and should move to sql-plugin-api
after refactoring. Currently there are still a lot of intertwined jdep dependencies with the shimmed area.
If a signature is inherently unshimmable we need to do more invasive refactoring.